src/sessionMessage.ts

```diff
@@ -0,0 +1,321 @@
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+import { H128 } from "codechain-sdk/lib/core/H128";
+import { H256 } from "codechain-sdk/lib/core/H256";
+import { H512 } from "codechain-sdk/lib/core/H512";
+
+const RLP = require("rlp");
+
+export default class Connection {
+
+}
+
+// Session message
+interface IBody {
+    toEncodeObject: () => Array<any> | number | string;
+    protocolId: () => number;
+}
+
+type Body = NodeIdRequest | NodeIdResponse | SecretRequest | SecretAllowed | SecretDenied | NonceRequest | NonceAllowed | NonceDenied;
+
+export class SessionMessage {
+    private version: number;
+    private seq: number;
+    private body: Body;
+
+    constructor(version: number, seq: number, body: Body) {
+        this.version = version;
+        this.seq = seq;
+        this.body = body;
+    }
+
+    setVersion(version: number) {
+        this.version = version;
+    }
+
+    setSeq(seq: number) {
+        this.seq = seq;
+    }
+
+    setBody(body: Body) {
+        this.body = body;
+    }
+
+    getVersion() {
+        return this.version;
+    }
+
+    getSeq() {
+        return this.seq;
+    }
+
+    getBody() {
+        return this.body;
+    }
+
+    toEncodeObject(): Array<any> | number | string {
+        return [this.version, this.seq, this.body.protocolId(), this.body.toEncodeObject()];
+
+    }
+
+    rlpBytes(): Buffer {
+        return RLP.encode(this.toEncodeObject());
+    }
+}
+
+export class NodeIdResponse implements IBody {
+    private nodeid: NodeId;
+
+    constructor(obj?: INodeId) {
+        const ip = obj && obj.ip || "0.0.0.0";
+        const port = obj && obj.port || 3485;
+        this.nodeid = new NodeId(ip, port);
+    }
+
+    setNodeid(obj: INodeId) {
+        this.nodeid.setIp(obj.ip);
+        this.nodeid.setPort(obj.port);
+    }
+
+    getNodeid() {
+        return this.nodeid;
+    }
+
+    toEncodeObject(): Array<any> | number | string {
+        return this.nodeid.toEncodeObject();
+    }
+    
+    protocolId(): number {
+        return 2;
+    }
+}
+
+export class NodeIdRequest implements IBody {
+    private nodeid: NodeId;
+
+    constructor(obj?: INodeId) {
+        const ip = obj && obj.ip || "0.0.0.0";
+        const port = obj && obj.port || 3485;
+        this.nodeid = new NodeId(ip, port);
+    }
+
+    setNodeid(obj: INodeId) {
+        this.nodeid.setIp(obj.ip);
+        this.nodeid.setPort(obj.port);
+    }
+
+    getNodeid() {
+        return this.nodeid;
+    }
+
+    toEncodeObject(): Array<any> | number | string {
+        return this.nodeid.toEncodeObject();
+    }
+
+    protocolId(): number {
+        return 1;
+    }
+}
+
+interface INodeId {
+    ip: string;
+    port: number;
+}
+
+class NodeId {
+    private ip: string; // x.x.x.x
+    private port: number; // x
+
+    constructor(ip: string, port: number) {
+        this.ip = ip;
+        this.port = port;
+        
+    }
+        setIp(ip: string) {

```

Comment:
Wrong indentation here. Please use the formatting tool. We use Prettier.

Reply:
I resolved it.